### PR TITLE
HOCS-3718 Extend Extraction Capability to Extract data from case_dead…

### PR DESCRIPTION
**Pull request by agraham19**
…line_extension table

* Add endpoint to produce Extensions' extract for a given case type

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java [hunk 1]
```diff
@@ -333,6 +337,54 @@ void correspondentExport(LocalDate from, LocalDate to, OutputStreamWriter output
...
+                    outputWriter.flush();
+                } catch (IOException e) {
+                    log.error("Unable to get record for audit {} for reason {}", audit.getUuid(), e.getMessage(), value(LogEvent.EVENT, CSV_EXPORT_FAILURE));
+                }
+            });
+            log.info("Export EXTENSION to CSV Complete", value(EVENT, CSV_EXPORT_COMPETE));
+        }
+    }
+
+    private String[] getExtensionAuditPayloadRow(AuditData audit, final ZonedDateTimeConverter zonedDateTimeConverter) throws IOException {
```
Comment (gary-newbolt) [inline src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java:374]: Can you change the name of the method to match the pattern used throughout this module please? i.e. `parseExtensionAuditPayload`

File: src/test/java/uk/gov/digital/ho/hocs/audit/export/AuditExportServiceTest.java [hunk 2]
```diff
@@ -466,6 +512,16 @@ public void verifyHeadersAreSubstitutedWithCorrespondentExtract() throws IOExcep
         verify(passThroughHeaderConverter, times(1)).substitute(anyList());
     }
 
+    @Test
+    public void verifyHeadersAreSubstitutedWithExtractExtract() throws IOException {
```
Comment (gary-newbolt) [inline src/test/java/uk/gov/digital/ho/hocs/audit/export/AuditExportServiceTest.java:516]: Should the method name on this test be `ExtensionExtract`?

**Review by gary-newbolt — CHANGES_REQUESTED** [review]
looks ok... It's just a few consistency issues on naming to address. And also a failing test in the pipeline, if they run ok locally, try running the pipeline again first.

File: src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java (hunk 1 above)
Comment (agraham19) [reply src/main/java/uk/gov/digital/ho/hocs/audit/export/ExportService.java]: I have addressed this

File: src/test/java/uk/gov/digital/ho/hocs/audit/export/AuditExportServiceTest.java (hunk 2 above)
Comment (agraham19) [reply src/test/java/uk/gov/digital/ho/hocs/audit/export/AuditExportServiceTest.java]: I have addressed this

**Review by gary-newbolt — APPROVED** [review]
lgtm

**Pull request merged**